internal/store/certificatesigningrequest.go

```diff
@@ -164,7 +164,6 @@ func (s *CertificateSigningRequestStore) createOrUpdate(orgId uuid.UUID, resourc
 		return nil, false, false, err
 	}
 	certificatesigningrequest.OrgID = orgId
-	certificatesigningrequest.Status = nil
```

Comment:
I must say i have no idea why it was here but i hope it doesn't break other things.

Reply:
This function is based on the enrollmentrequest version of the same functionality, and those requests must use Status in a different way (not sure why they'd want to set it to nil there). Nothing is relying on this code yet for the CSR functionality so I don't think that's an issue.

Reply:
Status was being set to nil here because Create, Update and CreateOrUpdate are user-facing, and the user shouldn't update Status.  That's why we have a separate UpdateStatus method.  I wouldn't mind adding 
a list of conditions as a parameter to UpdateStatus.

Reply:
I can redo it, please hold

Reply:
It *looks* like `UpdateStatus()` shouldn't need any more parameters as it already takes the whole CSR resource and uses its status